feat: add C implementation for `stats/base/dists/normal/logcdf` (#3988)
stdlib-js/stdlib: lib/node_modules/@stdlib/stats/base/dists/normal/logcdf/README.md

<!-- /.examples -->

<!-- C interface documentation. -->

* * *

<section class="c">

## C APIs

<!-- Section to include introductory text. Make sure to keep an empty line after the intro `section` element and another before the `/section` close. -->

<section class="intro">

</section>

<!-- /.intro -->

<!-- C usage documentation. -->

<section class="usage">

### Usage

```c
#include "stdlib/stats/base/dists/normal/logcdf.h"
```

> #### stdlib_base_dists_normal_logcdf( x, a, b )

Reviewer: ```suggestion
#### stdlib_base_dists_normal_logcdf( x, mu, sigma )
```

Evaluates the logarithm of the cumulative distribution function (logCDF) for a normal distribution.

```c
double out = stdlib_base_dists_normal_cdf( 2.0, 0.0, 1.0 );
// returns ~-0.023
```

The function accepts the following arguments:

-   **x**: `[in] double` input value.
-   **mu**: `[in] double` mean.
-   **sigma**: `[in] double` standard deviation.

```c
double stdlib_base_dists_normal_logcdf( cconst double x, const double mu, const double sigma );
```

</section>

<!-- /.usage -->

<!-- C API usage notes. Make sure to keep an empty line after the `section` element and another before the `/section` close. -->

<section class="notes">

</section>

<!-- /.notes -->

<!-- C API usage examples. -->

<section class="examples">

### Examples

```c
#include "stdlib/stats/base/dists/normal/logcdf.h"
#include <stdlib.h>
#include <stdio.h>

static double random_uniform( const double min, const double max ) {
    double v = (double)rand() / ( (double)RAND_MAX + 1.0 );
    return min + ( v*(max-min) );
}

int main( void ) {
    double x;
    double mu;
    double sigma;
    double y;

>     int i;

Reviewer: By convention we order variable declarations by length:

```suggestion
    double sigma;
    double mu;
    double x;
    double y;
    int i;
```

<!-- /.examples -->

<!-- Section to include cited references. If references are included, add a horizontal rule *before* the section. Make sure to keep an empty line after the `section` element and another before the `/section` close. -->

<!-- C interface documentation. -->

* * *

<section class="c">

## C APIs

<!-- Section to include introductory text. Make sure to keep an empty line after the intro `section` element and another before the `/section` close. -->

<section class="intro">

</section>

<!-- /.intro -->

<!-- C usage documentation. -->

<section class="usage">

### Usage

```c
#include "stdlib/stats/base/dists/normal/logcdf.h"
```

#### stdlib_base_dists_normal_logcdf( x, mu, sigma )

> Evaluates the logarithm of the cumulative distribution function (logCDF) for a normal distribution.

Reviewer: ```suggestion
Evaluates the natural logarithm of the [cumulative distribution function][cdf] (CDF) for a [normal][normal-distribution] distribution with parameters `mu` (mean) and `sigma` (standard deviation).
```
